tmpl/blog/announcing-mirage-35-release.md

```diff
@@ -0,0 +1,107 @@
+It requires a requested `size` argument to be passed, and a fill function which is called with an allocated buffer, that satisfies the backend demands. The `fill` function is supposed to write to the buffer, and return the length of the frame to be send out. It can neither error (who should handle such an error anyways?), nor is it in the IO monad. The `fill` function should not save any references to the buffer, since this is the network device's memory, and may be reused. The `writev` function has been removed.
+
+The [Ethernet layer](https://github.com/mirage/mirage-protocols) does encapsulation and decapsulation now. Its `write` function has the following signature:
+`val write: t -> ?src:macaddr -> macaddr -> Ethernet.proto -> ?size:int -> (buffer -> int) -> (unit, error) result io`.
+It fills in the Ethernet header with the given source address (defaults to the device's own MAC address) and destination address, and Ethernet protocol. The `size` argument is optional, and defaults to the MTU. The `buffer` that is passed to the `fill` function is usable from offset 0 on. The Ethernet header is not visible at higher layers.
+
+The IP layer also embeds a revised `write` signature:
+`val write: t -> ?fragment:bool -> ?ttl:int -> ?src:ipaddr -> ipaddr -> Ip.proto -> ?size:int -> (buffer -> int) -> buffer list -> (unit, error) result io`.
+This is similar to the Ethernet signature - it writes the IPv4 header and sends a packet. It also supports fragmentation (including setting the do-not-fragment bit for path MTU discovery) -- whenever the payload is too big for a single frame, it is sent as multiple fragmented IPv4 packets. Additionally, setting the time-to-live is now supported, meaning we now can implement traceroute!
+The API used to include two functions, `allocate_frame` and `write`, where only buffers allocated by the former should be used in the latter. This has been combined into a single function that takes a fill function and a list of payloads. This change is for maximum flexibility: a higher layer can either construct its header and payload, and pass it to `write` as payload argument (the `buffer list`), which is then copied into the buffer(s) allocated by the network device, or the upper layer can provide the callback `fill` function to assemble its data into the buffer allocated by the network device, to avoid copying. Of course, both can be used - the outgoing packet contains the IPv4 header, and possibly the buffer until the offset returned by `fill`, and afterwards the payload.
+
+The TCP implementation has [preliminary keepalive support](https://github.com/mirage/mirage-tcpip/pull/338).
+
+### [Solo5](https://github.com/solo5/solo5)
+
+- MirageOS 3.0.0 used the 0.2.0 release of solo5
+- The `ukvm` target was renamed to `hvt`, where `solo5-hvt` is the monitoring process
+- Support for [FreeBSD bhyve](http://bhyve.org/) and [OpenBSD VMM](https://man.openbsd.org/vmm.4) hypervisor (within the hvt target)
```

Comment:
These new features seem like a really big deal for users, and hidden away somewhat.  We could trumpet "new platform support" for FreeBSD and OpenBSD and Linux/arm64 support right at the top?

Reply:
feel free to directly push to this branch!